kratos/python_scripts/restart_utility.py

```diff
@@ -170,6 +191,47 @@ def CreateOutputFolder(self):
                 os.makedirs(folder_path)
             self.model_part.GetCommunicator().GetDataCommunicator().Barrier()
 
+    def GetRestartFiles(self):
```

Comment:
In here may be you can try the following
```python
file_name_data_collector = KratosMultiphysics.FileNameDataCollector(self.model_part, self.__GetFileNamePattern() + ".rest", {})
file_name_data_list = self.file_name_data_collector.GetFileNameDataList()
sorted_file_name_data_list = self.file_name_data_collector.GetSortedListOfFileNameData(file_name_data_list, ["<step>", "<time>", "<rank>"])

# now you have the list of restart files sorted first with step, then time finally the rank
# now you can generate restart_files map.
```

With this you don't have to use "__IsRestartFile" or "is_file()", all of them will be taken care by the `FileNameDataCollector`. Remember, now `self.__GetFilenamePattern()` should also include the total path as well

Reply:
Thanks for the hints, I made the updates